src/Notepads/Controls/TextEditor/TextEditor.xaml.cs

```diff
@@ -857,37 +908,50 @@ public void HideFindAndReplaceControl()
         private void FindAndReplaceControl_OnFindAndReplaceButtonClicked(object sender, FindAndReplaceEventArgs e)
         {
             TextEditorCore.Focus(FocusState.Programmatic);
+            InitiateFindReplace(e);
+
+            // In case user hit "enter" key in search box instead of clicking on search button or hit F3
+            // We should re-focus on FindAndReplaceControl to make the next search "flows"
+            if (!(sender is Button))
+            {
+                FindAndReplaceControl.Focus(e.FindAndReplaceMode);
+            }
+        }
+
+        private void InitiateFindReplace(FindAndReplaceEventArgs findAndReplaceEventArgs)
+        {
+            if (string.IsNullOrEmpty(findAndReplaceEventArgs.SearchText)) return;
+
             bool found = false;
-            string regexError = null;
+            bool regexError = false;
 
-            switch (e.FindAndReplaceMode)
+            if (FindAndReplacePlaceholder?.Visibility == Visibility.Visible)
+                _searchCache = new SearchCache(findAndReplaceEventArgs.SearchText, findAndReplaceEventArgs.MatchCase, findAndReplaceEventArgs.MatchWholeWord, findAndReplaceEventArgs.UseRegex);
```

Comment:
Is the _searchCache intended to carry current search settings to all newly added editors?  If yes, we might need to not do it for now. Reason: there is no UI to show what search options user has been made. I am think about the design here, will let you know once I come up with one.

Reply:
> Is the _searchCache intended to carry current search settings to all newly added editors?  If yes, we might need to not do it for now. Reason: there is no UI to show what search options user has been made. I am think about the design here, will let you know once I come up with one.

No every editor will have thier separate search results in `_searchcache` and for new editors it would have empty search string with default options.